xbmc/pvr/channels/PVRChannelGroup.cpp

```diff
@@ -287,18 +286,18 @@ const PVRChannelGroupMember& CPVRChannelGroup::GetByUniqueID(const std::pair<int
 
 std::shared_ptr<CPVRChannel> CPVRChannelGroup::GetByUniqueID(int iUniqueChannelId, int iClientID) const
 {
-  return GetByUniqueID(std::make_pair(iClientID, iUniqueChannelId)).channel;
+  return GetByUniqueID(std::make_pair(iClientID, iUniqueChannelId))->channel;
 }
 
 std::shared_ptr<CPVRChannel> CPVRChannelGroup::GetByChannelID(int iChannelID) const
 {
   std::shared_ptr<CPVRChannel> retval;
   CSingleLock lock(m_critSection);
 
-  for (PVR_CHANNEL_GROUP_MEMBERS::const_iterator it = m_members.begin(); !retval && it != m_members.end(); ++it)
+  for (const auto& memberPair : m_members)
```

Comment:
This changes the logic! Original code escapes fom the loop as soon as `retval` is non-zero!

I suggest to get rid of `retval`and early return directly from the loop on match.

Reply:
I just I had correctly skipped all of these!